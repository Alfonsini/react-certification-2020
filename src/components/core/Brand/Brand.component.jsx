import React from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlay } from '@fortawesome/free-solid-svg-icons';

import { StyledUl, StyledLi, StyledLink, StyledLinkTitle } from './Brand.styles';

function Brand(props) {
  return (
    <StyledUl role="list">
      <StyledLi>
        <StyledLink to="/" expanded={props.expanded}>
          <FontAwesomeIcon icon={faPlay} />
          <StyledLinkTitle textTruncate>Player</StyledLinkTitle>
        </StyledLink>
      </StyledLi>
    </StyledUl>
  );
}

export default Brand;
