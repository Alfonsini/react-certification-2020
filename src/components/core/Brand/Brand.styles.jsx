import styled from 'styled-components';

const StyledUl = styled.ul`
  font-size: 1.1rem;
  font-family: 'Montserrat', Helvetica, Arial, serif;
  font-weight: 400;
  overflow-y: hidden;
  overflow-x: hidden;
  list-style: none;
  margin: 0;
  padding: 0;
  padding-bottom: 0.75rem;

  margin-top: 1rem;
`;

const StyledLi = styled.li`
  position: relative;
  white-space: nowrap;
  margin-bottom: 18px;
`;

const StyledLink = styled.a`
  color: #7367f0;
  margin: 0 15px;
  padding: 10px 15px 10px 15px;
  line-height: 1.45;
  text-overflow: ellipsis;
  overflow: hidden;
  align-items: center;
  ${(props) => !props.expanded && `width: 50px`};
  ${(props) => props.expanded && `width: calc(260px - 35px)`};
  display: flex;

  svg {
    margin-right: 1.1rem;
  }
`;

const StyledShadowBottom = styled.div`
  display: none;
  position: absolute;
  z-index: 2;
  height: 50px;
  width: 100%;
  pointer-events: none;
  filter: blur(5px);
`;

const StyledLinkTitle = styled.span`
  ${(props) =>
    props.textTruncate &&
    `
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    `}
`;

export { StyledUl, StyledLi, StyledLink, StyledShadowBottom, StyledLinkTitle };
