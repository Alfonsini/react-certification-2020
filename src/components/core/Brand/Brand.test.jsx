import React from 'react';
import { render, screen } from '../../../test/testUtils';

import Brand from '.';

describe('Brand', () => {
  test('sidebar not expanded', () => {
    render(<Brand />);

    const aside = screen.getByRole('list');
    expect(aside).toMatchSnapshot();
  });

  test('sidebar expanded', () => {
    render(<Brand expanded />);

    const aside = screen.getByRole('list');
    expect(aside).toMatchSnapshot();
  });
});

