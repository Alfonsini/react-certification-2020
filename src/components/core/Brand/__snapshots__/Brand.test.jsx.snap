// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand sidebar expanded 1`] = `
<ul
  class="sc-bkzZxe emqQXN"
  role="list"
>
  <li
    class="sc-idOhPF eNZqtm"
  >
    <a
      class="sc-dIUggk dxdciD"
      to="/"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-play fa-w-14 "
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          fill="currentColor"
        />
      </svg>
      <span
        class="sc-dmlrTW jbYabi"
      >
        Player
      </span>
    </a>
  </li>
</ul>
`;

exports[`Brand sidebar not expanded 1`] = `
<ul
  class="sc-bkzZxe emqQXN"
  role="list"
>
  <li
    class="sc-idOhPF eNZqtm"
  >
    <a
      class="sc-dIUggk ExmPp"
      to="/"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-play fa-w-14 "
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          fill="currentColor"
        />
      </svg>
      <span
        class="sc-dmlrTW jbYabi"
      >
        Player
      </span>
    </a>
  </li>
</ul>
`;
